engine/classes/Elgg/I18n/Translator.php

```diff
@@ -42,6 +42,13 @@ function translate($message_key, $args = array(), $language = "") {
 	
 		static $CURRENT_LANGUAGE;
 	
+		if (!is_string($message_key) || strlen($message_key) < 1) {
+			_elgg_services()->logger->warn(
+				'$message_key needs to be a string in ' . __METHOD__ . '(), ' . gettype($message_key) . ' provided'
```

Comment:
with this the error message is
> WARNING: $message_key needs to be a string in Elgg\I18n\Translator::translate(), array provided

which makes it a bit more readable, so there's no dubble backet